### PR TITLE
feat: implement write interfaces for image cell system contract

### PR DESCRIPTION
**Pull request by felicityin**
**What this PR does / why we need it**:

This PR implements write interfaces for image cell system contract

**Which issue(s) this PR fixes**:

Fixes #

**Which docs this PR relation**:

Ref https://github.com/axonweb3/axon/discussions/896

**Which toolchain this PR adaption**:

No Breaking Change

**Special notes for your reviewer**:

NIL

**CI Description**

| CI Name                                   | Description                                                     |
| ----------------------------------------- | --------------------------------------------------------------- |
| *Chaos CI*                                | Test the liveness and robustness of Axon under terrible network condition    |
| *Cargo Clippy*                            | Run `cargo clippy --all --all-targets --all-features`      |
| *Coverage Test*                           | Get the unit test coverage report                             |
| *E2E Test*                                | Run end-to-end test to check interfaces                         |
| *Code Format*                             | Run `cargo +nightly fmt --all -- --check` and `cargo sort -gwc`     |
| *Web3 Compatible Test*                    | Test the Web3 compatibility of Axon                               |
| *v3 Core Test*                            | Run the compatibility tests provided by Uniswap V3             |
| *OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19* | Run the compatibility tests provided by OpenZeppelin           |

**CI Usage**

> Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [ ] Chaos CI
- [x] Cargo Clippy
- [ ] Coverage Test
- [ ] E2E Tests
- [x] Code Format
- [x] Unit Tests
- [ ] Web3 Compatible Tests
- [ ] OCT 1-5 And 12-15
- [ ] OCT 6-10
- [ ] OCT 11
- [ ] OCT 16-19
- [ ] v3 Core Tests

### CODE REVIEW

File: core/executor/src/system_contract/image_cell/abi/README.md [hunk 1]
```diff
@@ -1,14 +1,15 @@
 # Generate `ImageCell.abi.json`
 
-`ImageCell.abi.json` is generated by [hardhat](https://hardhat.org/tutorial/writing-and-compiling-contracts).
+API is generated by [hardhat](https://hardhat.org/tutorial/writing-and-compiling-contracts).
 
 ```
 cd ../contract
+yarn
```
Comment (wenyuanhust) [inline core/executor/src/system_contract/image_cell/abi/README.md:7]: I just follow the steps here 
> https://hardhat.org/tutorial/creating-a-new-hardhat-project
No need to execute `yarn`.

Comment (felicityin) [reply core/executor/src/system_contract/image_cell/abi/README.md:7]: `yarn` will install all dependencies. It only needs to be executed once.

File: core/executor/src/system_contract/image_cell/mod.rs [hunk 2]
```diff
@@ -1 +1,204 @@
...
+pub use error::{ImageCellError, ImageCellResult};
+pub use store::{cell_key, header_key, CellInfo, CellKey, HeaderKey};
+use store::{get_block_number, get_cell, get_header};
+use trie_db::RocksTrieDB;
+
+static ALLOW_READ: AtomicBool = AtomicBool::new(false);
+static TRIE_DB: OnceCell<Arc<RocksTrieDB>> = OnceCell::new();
+
+lazy_static::lazy_static! {
+    static ref MPT_ROOT_KEY: H256 = H256::default();
```
Comment (KaoImin) [inline core/executor/src/system_contract/image_cell/mod.rs]:
```suggestion
    static ref CELL_ROOT_KEY: H256 = Hasher::digest("cell_mpt_root");
```

File: core/executor/src/system_contract/image_cell/mod.rs [hunk 3]
```diff
@@ -1 +1,204 @@
...
+
+fn get_mpt<B: Backend + ApplyBackend>(backend: &B) -> ImageCellResult<MPTTrie<RocksTrieDB>> {
+    let trie_db = match TRIE_DB.get() {
+        Some(db) => db,
+        None => return Err(ImageCellError::TrieDbNotInit),
+    };
+
+    let root = backend.storage(ImageCellContract::ADDRESS, *MPT_ROOT_KEY);
+
+    if root == *H256_DEFAULT {
```
Comment (KaoImin) [inline core/executor/src/system_contract/image_cell/mod.rs]:
```suggestion
    if root == H256::default() {
```

File: core/executor/src/system_contract/image_cell/mod.rs [hunk 4]
```diff
@@ -1 +1,204 @@
...
+pub use store::{cell_key, header_key, CellInfo, CellKey, HeaderKey};
+use store::{get_block_number, get_cell, get_header};
+use trie_db::RocksTrieDB;
+
+static ALLOW_READ: AtomicBool = AtomicBool::new(false);
+static TRIE_DB: OnceCell<Arc<RocksTrieDB>> = OnceCell::new();
+
+lazy_static::lazy_static! {
+    static ref MPT_ROOT_KEY: H256 = H256::default();
+    static ref H256_DEFAULT: H256 = H256::default();
```
Comment (KaoImin) [inline core/executor/src/system_contract/image_cell/mod.rs]: useless code

**Review by KaoImin — APPROVED** [review]

**Pull request merged**